### PR TITLE
ci: upgraded ruby/setup-ruby action from version v1.152.0 to v1.178.0

### PR DESCRIPTION
**Pull request by desusai7**
### 📋 Changes

* upgraded ruby/setup-ruby action from version v1.152.0 to v1.178.0 to support ruby version `3.0.7`
* Added codecov token as a secret to workflows to get rid of rate limiting error.

### 📎 References

https://github.com/auth0/JWTDecode.swift/actions/runs/9598725630/job/26470757827

### CODE REVIEW

**Review by Widcket — DISMISSED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,8 +13,17 @@ concurrency:
     cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
 
 jobs:
+    authorize:
+      name: Authorize
+      environment: ${{ github.event.pull_request.head.repo.fork && 'external' || 'internal' }}
+      runs-on: ubuntu-latest
+      steps:
+        - run: true
```
Comment (frederikprijck) [inline .github/workflows/main.yml:21]: The authorize step is required when u use `pull_request_target` in the `on` on line 3. If u use `pull_request`, authorize has no purpose.

You should ask your self:

- should codecov work for forks? 
  - if yes => use pull_request_target and authorize
  - if no => keep pull_request and drop authorize.

**Review by frederikprijck — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (desusai7) [reply .github/workflows/main.yml:21]: Hi @frederikprijck, 

Thank you very much for the detailed explanation on the differences between the triggers `pull_request` and `pull_request_target`.

We would like to have codecov working for forks as well but it needs to be authorized first, hence proceeded with `pull_request_target`.

Pushed the changes accordingly, please take a look again

Comment (desusai7) [reply .github/workflows/main.yml:21]: On further discussion with @frederikprijck, I've decided to remove the code coverage check for the PR's raised from forks, hence changing it back to `pull_request` and dropping `authorize`

File: README.md [hunk 2]
```diff
@@ -76,7 +76,7 @@ Then, run `carthage bootstrap --use-xcframeworks`.
 import JWTDecode
 ```
 
-2. Decode the token
+2. Decode the token using the code snippet shared below:
```
Comment (Widcket) [inline README.md:79]: Was this added for CI purposes or is it an intended change?

Comment (Widcket) [reply README.md:79]: (for triggering a build, I mean)

Comment (desusai7) [reply README.md:79]: Yes, will revert them

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ concurrency:
 
 jobs:
     test:
-        name: Test on ${{ matrix.platform.os }} using Xcode ${{ matrix.xcode }}
+        name: Test on ${{ matrix.platform.os }} using Xcode ${{ matrix.xcode }}s
```
[the remark this answers is not in the document]
Comment (desusai7) [reply .github/workflows/main.yml]: Updated

**Review by Widcket — APPROVED** [review]

**Review by Widcket — APPROVED** [review]

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -15,6 +15,7 @@ concurrency:
 jobs:
     test:
         name: Test on ${{ matrix.platform.os }} using Xcode ${{ matrix.xcode }}
+        environment: ${{ github.event.pull_request.head.repo.fork && 'external' || 'internal' }}
```
Comment (frederikprijck) [inline .github/workflows/main.yml:18]: Why is this needed?

Comment (desusai7) [reply .github/workflows/main.yml]: I had added it previously, when we have introduced two different environments for secrets as part of using the `authorize` step. 

Removed it now, as we are no longer using `authorize`, please take a look again.

**Review by frederikprijck — APPROVED** [review]
